Snowfall

Hibernathan1212 opened the pull request:
# BrowserBuddy Extension Submission

## Project Details

1. **What is the name of your project?** - Snowfall

2. **How many hours did you spend developing this?**  - ~2-3 hours
   _(Give your best estimate)_

3. **What problem does your extension solve, or what makes it unique?**  - Simple and minimalist design for toggling snow on your browser and all webpages. Additionally shows a countdown until Christmas on the popup
   _(Provide a brief description of the functionality or innovation)_

4. **How many people have tested your extension?**  - 3, been tested on Chrome, Chromium, Firefox, and even other chromium based browsers such as Arc.
   _(We recommend at least 3)_

5. **Does your extension work in both Chrome and Firefox?**  - Yes works perfectly on both!
   _(Your extension must be compatible with both browsers)_

<img width="1624" alt="Screenshot 2024-12-27 at 8 52 13 PM" src="https://github.com/user-attachments/assets/209a8326-8d44-4c45-bd86-9aadc5403aa6" />

<img width="1624" alt="Screenshot 2024-12-27 at 8 53 59 PM" src="https://github.com/user-attachments/assets/78afd9f0-4d4a-4729-b34b-cc602b7cab88" />

---

## Additional Comments

(Feel free to provide any additional context or notes here)

JosiasAurel reviewed (changes requested):
Hey. Can you try something different or another feature? Someone already made an extension that adds snowfall to websites.

Hibernathan1212 commented:
I'm sorry, I didn't see previous extension. My extension also has the Christmas countdown. Do you have any suggestions for other features I could add that I could get done by the 31st?

JosiasAurel commented:
> I'm sorry, I didn't see previous extension. My extension also has the Christmas countdown. Do you have any suggestions for other features I could add that I could get done by the 31st?

Try adding interactive garlands like those on [Festivitas](https://simonbs.gumroad.com/l/festivitas)?

Hibernathan1212 commented:
I have added interactive garlands that automatically move out of the way when accessing something from the top of the screen. 

<img width="2672" alt="Screenshot 2024-12-30 at 2 07 15 PM" src="https://github.com/user-attachments/assets/7a0b059c-e412-4d69-9848-6dca54dcb106" />


https://github.com/user-attachments/assets/3e54903d-4e22-41a5-a47b-b63c2867d60a

JosiasAurel commented on submissions/Snowfall/manifest.json:
```diff
@@ -0,0 +1,43 @@
+{
+  "manifest_version": 2,
```
your extension must be using manifest version 3

JosiasAurel replied:
I think it'll also be nice if the garlands were lighting up

Hibernathan1212 replied:
How do you make firefox work with manifest v3 background service workers? I'm not entirely sure how I could do that. Also aren't garlands and lights seperate things? I'll try adding lights to them.

Hibernathan1212 replied:
Can I have two different versions for chrome and for firefox, one using manifest V3 and the other V2 respectively?

Hibernathan1212 commented:
I've added some small flickering lights to the garlands. I've also split the Firefox and Chrome extensions into different files as they now have slightly different code to make them both work with manifest V3.

JosiasAurel reviewed (approved):
Nice. Please fill [this form](https://airtable.com/appgtKZwZBoOeFU7O/pagyeS51zr2kC1rp0/form) to get your grant.
